tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/mailet/CalDavCollect.java

```diff
@@ -0,0 +1,145 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.tmail.mailet;
+
+import static com.linagora.tmail.dav.DavClient.CALENDAR_PATH;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+
+import jakarta.inject.Inject;
+import jakarta.mail.MessagingException;
+
+import org.apache.james.core.Username;
+import org.apache.mailet.AttributeName;
+import org.apache.mailet.AttributeUtils;
+import org.apache.mailet.AttributeValue;
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailetException;
+import org.apache.mailet.base.GenericMailet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linagora.tmail.dav.DavCalendarObject;
+import com.linagora.tmail.dav.DavClient;
+import com.linagora.tmail.dav.DavUser;
+import com.linagora.tmail.dav.DavUserProvider;
+import com.linagora.tmail.dav.EventUid;
+
+import net.fortuna.ical4j.model.Calendar;
+import net.fortuna.ical4j.model.Property;
+import net.fortuna.ical4j.model.component.VEvent;
+import net.fortuna.ical4j.model.property.Uid;
+import reactor.core.publisher.Mono;
+
+public class CalDavCollect extends GenericMailet {
+    public static final class CalendarClients {
```

Comment:
Is that really necessary? Can't we just inject directly davClient and davUserProvider? If missing cause not using OpenPaasModule it will throw directly at Tmail start anyways? I believe other mailets function like this..?

Reply:
The code seems to create a lazy bean.
It will look more pretty:

![image](https://github.com/user-attachments/assets/094e4b93-4877-4897-bac5-09bb08163835)